src/main/java/io/github/kawamuray/wasmtime/Func.java

```diff
@@ -1,16 +1,53 @@
 package io.github.kawamuray.wasmtime;
 
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
-import java.util.Optional;
-
 import lombok.AccessLevel;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.experimental.Accessors;
 import lombok.extern.slf4j.Slf4j;
 
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A WebAssembly function which can be called.
+ *
+ * This type can represent either an exported function from a WebAssembly module or a host-defined function which can
+ * be used to satisfy an import of a module. Func and can be used to both instantiate an Instance as well as be
+ * extracted from an Instance.
+ *
+ * A Func "belongs" to the store that it was originally created within. Operations on a Func only work with the store
+ * it belongs to, and if another store is passed in by accident then methods will panic.
+ *
+ * <h1>Func and async</h1>
+ * Functions from the perspective of WebAssembly are always synchronous. You might have an async function in Rust,
```

Comment:
Here has the same problem too. These statements about async function is for Rust developers and isn't suitable for Java developers.
I don't think we need fully copy of the document for Java side (at least for the time being) as people can navigate to the original wasmtime API if they want detailed documents. So why don't we shrink this document to explain least information and perhaps link to the wasmtime's Func implementation?

Reply:
I also changed the original documentation slightly to make it concise and clear.

I feel conflicted though. Being able to see the full intention of the Rust Library is also great from Java. Albeit not technically correct.
Guess it's a tooling issue though, haha :).